libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1613,6 +1613,13 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 	_operation.annotation().isPure = !modifying && *_operation.subExpression().annotation().isPure;
 	_operation.annotation().isLValue = false;
 
+	if (op == Token::Delete && subExprType->containsNestedMapping())
```

Comment:
Is it more convenient to move this check inside `Type::unaryOperatorResult`? We would only need it on reference types.

Reply:
Could could move it, but the error reporter is not available in the type and in this particular visit, the string part of the return value of that function is also not used for the error message

Reply:
I could change the call here to use the err message returned.. it would change a few.. like this:

```
syntaxTests/negation.sol: FAIL
  Contract:
    contract test {
        function f() public pure {
            int x;
            uint y = uint(-x);
            -y;
        }
    }

  Expected result:
    TypeError 4907: (97-99): Unary operator - cannot be applied to type uint256
  Obtained result:
    TypeError 4907: (97-99): Unary operator - cannot be applied to type uint256: Unary negation is only allowed for signed integers.


```

Reply:
That actually sounds good!